src/libs/ascent/runtimes/flow_filters/ascent_runtime_relay_filters.cpp

```diff
@@ -730,6 +755,112 @@ RelayIOSave::execute()
 
     Node *in = n_input.get();
 
+    // if we are doing overlink, we must enforce the 1-topo rule
+    if (protocol == "overlink")
+    {
+        // either users have specified a topology they want to use
+        if (params().has_path("topologies"))
+        {
+            const conduit::Node &tlist = params()["topologies"];
+            const int num_topos = tlist.number_of_children();
+            if (num_topos != 1)
+            {
+                ASCENT_ERROR("relay_io_save Overlink save requires a single topology; " <<
+                             num_topos << " topologies were requested.");
+            }
+        }
```

Comment:
@cyrush Can I assume params are the same across all ranks?

Reply:
yes, pre condition is that all ranks provide the same set of actions, so you don't need to handle that case.